numba/datamodel/models.py

```diff
@@ -208,9 +222,22 @@ def __init__(self, dmm, fe_type):
         be_type = ir.IntType(8).as_pointer()
         super(OpaqueModel, self).__init__(dmm, fe_type, be_type)
 
+
+@register_default(types.MemInfoPointer)
+class MemInfoModel(OpaqueModel):
+
+    def inner_types(self):
+        return self._dmm.lookup(self._fe_type.dtype).traverse_types()
+
+    def has_nrt_meminfo(self):
+        return True
+
     def get_nrt_meminfo(self, builder, value):
-        if self._fe_type == types.meminfo_pointer:
-            return value
+        for tp in self.inner_types():
+            if self._dmm.lookup(tp).has_nrt_meminfo():
+                raise NotImplementedError(
```

Comment:
Did you encounter something that require nested MemInfo?  I imagine other managed object to exist as reference (CPointer) inside other MemInfo.